tests/unit_tests/test_saas_adaptation.py

```
@@ -0,0 +1,765 @@
+# ===========================================================================
+
+
+class TestStorageCacheNoSingleton:
+    """Tests verifying get_storage_cache_instance() no longer uses a global singleton."""
+
+    def test_returns_new_instance_each_call(self):
+        """get_storage_cache_instance returns a distinct object on each call."""
+        from datus.storage.cache import StorageCache, get_storage_cache_instance
+
+        mock_config = MagicMock()
+        mock_config.rag_storage_path.return_value = "/tmp/test"
```

Ruff (0.15.6)

[error] Probable insecure usage of temporary file or directory: "/tmp/test"

(S108)